packages/opentelemetry-resources/src/platform/node/detectors/AwsBeanstalkDetector.ts

```diff
@@ -42,11 +42,23 @@ class AwsBeanstalkDetector implements Detector {
         [SERVICE_RESOURCE.VERSION]: parsedData.version_label,
         [SERVICE_RESOURCE.INSTANCE_ID]: parsedData.deployment_id,
       });
-    } catch (e) {
-      config.logger.debug(`AwsEc2Detector failed: ${e.message}`);
+    } catch (err) {
+      config.logger.debug(`AwsEc2Detector failed: ${err.message}`);
       return Resource.empty();
     }
   }
+
+  private async _getData(): Promise<string> {
```

Comment:
There is an officially supported method to do this in node https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original

Reply:
That looks really cool, Let me do it!